### PR TITLE
Feat/232 add ci document cypress test

### PR DESCRIPTION
**Pull request by broulaye**
#### What do these changes do/fix?
Add CI Document to example app and add cypress test

#### How do you test/verify these changes?
Run Cypress test

#### Have you documented your changes (if necessary)?
Yes

#### Are there any breaking changes included in this pull request?
No

https://zenhub.ibm.com/app/workspaces/discovery-unified-5728bf39d2c20f776e05d0ef/issues/watson-discovery/docviz-squad-issue-tracker/232

### CODE REVIEW

File: examples/discovery-search-app/src/App.js [hunk 1]
```diff
@@ -3,8 +3,9 @@ import DiscoveryV2 from 'ibm-watson/discovery/v2';
 import { NoAuthAuthenticator } from 'ibm-watson/auth';
 import './app.scss';
 import { settings } from 'carbon-components';
-import { Button } from 'carbon-components-react';
+import { Button, Tabs, Tab } from 'carbon-components-react';
 import Close from '@carbon/icons-react/lib/close/16';
+import cx from 'classnames';
```
Comment (maniax89) [inline examples/discovery-search-app/src/App.js]: make sure to add `classnames` as a dependency for the example app

Comment (broulaye) [reply examples/discovery-search-app/src/App.js]: Good catch

**Comment by broulaye** [comment]
@jhpedemonte This is ready for a re-review

**Comment by broulaye** [comment]
@maniax89 This is ready for a re-review

File: examples/discovery-search-app/cypress/integration/passages/spec.ts [hunk 2]
```diff
@@ -90,5 +90,27 @@ describe('Passage Results', () => {
...
+    describe('and clicking on "CI Document" for a result', () => {
+      beforeEach(() => {
+        cy.get('button[data-testid="search-result-element-preview-button"]')
+          .contains('View table in document')
+          .click();
+        cy.findByText('CI');
+        cy.findByText('CI').click();
+      });
+
+      it('verify that CI Document contains filters', () => {
```
Comment (ColbyJohnIBM) [inline examples/discovery-search-app/cypress/integration/passages/spec.ts]: These messages should describe what you expect the application to do, not what the test should do (if that makes sense). (e.g. 'displays the filters')

Comment (broulaye) [reply examples/discovery-search-app/cypress/integration/passages/spec.ts]: make sense

File: examples/discovery-search-app/cypress/integration/passages/spec.ts [hunk 3]
```diff
@@ -90,5 +90,27 @@ describe('Passage Results', () => {
       cy.get('@combinedResult').contains('View passage in document');
       cy.get('@combinedResult').contains('View table in document');
     });
+
+    describe('and clicking on "CI Document" for a result', () => {
```
Comment (jhpedemonte) [inline examples/discovery-search-app/cypress/integration/passages/spec.ts]: There's a section above: 

`describe('and clicking on "View passage in document" for a result', () => {`

where this code should live. Between the `navigate to` and `close` tests. That is, create another nested `describe` for clicking on the CI tab and running these two tests.

File: examples/discovery-search-app/src/App.js [hunk 4]
```diff
@@ -85,9 +88,20 @@ function SearchPage() {
...
 
+  const tabs = [
+    {
+      name: 'Document',
+      Component: DocumentPreview
+    }
+  ];
+
+  if (canRenderCIDocument(document)) {
+    tabs.push({ name: 'CI', Component: CIDocument });
```
Comment (jhpedemonte) [inline examples/discovery-search-app/src/App.js]: I know it's long, but I think it would be better to have this as `Content Intelligence` instead of `CI`, which may not be easily recognizable by users.

**Comment by broulaye** [comment]
@jhpedemonte This is ready for a re-review

File: examples/discovery-search-app/cypress.json [hunk 5]
```diff
@@ -1,4 +1,5 @@
 {
   "baseUrl": "http://localhost:3000/",
-  "supportFile": "cypress/support/index.ts"
+  "supportFile": "cypress/support/index.ts",
+  "defaultCommandTimeout": 30000
```
Comment (jhpedemonte) [inline examples/discovery-search-app/cypress.json]: Why is this necessary?

Comment (broulaye) [reply examples/discovery-search-app/cypress.json]: Because otherwise some cypress tests timeout before the page had the time to render.

Comment (jhpedemonte) [reply examples/discovery-search-app/cypress.json]: Then perhaps we need to either fix our tests or our code. I believe the default is `10000` (10s) -- that should be more than enough time. Please open an issue to fix this, so we can remove this line.

Also, @H-Fin created a smaller contract in order to cut down on how long tests take. Are we using that here?

Comment (broulaye) [reply examples/discovery-search-app/cypress.json]: Probably not, since this was merged before that I believe. I can create an issue to fix this
